django/db/models/sql/compiler.py

```diff
@@ -160,7 +160,10 @@ def get_group_by(self, select, order_by):
         expressions = self.collapse_group_by(expressions, having_group_by)
 
         for expr in expressions:
-            sql, params = self.compile(expr)
+            try:
+                sql, params = self.compile(expr)
+            except EmptyResultSet:
+                continue
```

Comment:
This is to support this particular test https://github.com/django/django/blob/a2e580acf62a3de4f9108cc1af061a02b94b1064/tests/aggregation/tests.py#L2056-L2061